src/Handlers/ResultHandler.php

```diff
@@ -115,15 +115,13 @@ public function processResult(
             // Get MD and PaRes to be validated
             $md = $request->query->get('MD');
             $paRes = $request->query->get('PaRes');
-
-            if (empty($md) || empty($paRes)) {
-                throw new PaymentException('MD and/or PaRes parameter is missing from the redirect request');
-            }
+            $redirectResult = $request->query->get('redirectResult');
 
             // Construct the details object for the paymentDetails request
             $details = [
                 'MD' => $md,
-                'PaRes' => $paRes
+                'PaRes' => $paRes,
```

Comment:
I think it would be better to only add to the details when the variable is not empty like:

```
if (!empty($paRes)) {
    $details['PaRes'] = $paRes;
}
```

Reply:
Done!